File: sources/2015/11/quick-check/qc-java/src/main/java/qc/Supervisor.java
package qc;

public class Supervisor {

    private Double AMOUNT_MAX_ACCEPTED = 1000d;
    private Double AMOUNT_MIN_CHECKED = 200d;

    public Loan process(final Loan loan) {
        if (loan.amount > AMOUNT_MAX_ACCEPTED) {
            return new Loan(State.REJECTED, loan.amount);
        }

        if (loan.amount < AMOUNT_MIN_CHECKED) {
            return new Loan(State.ACCEPTED, loan.amount);
        }

        return new Loan(State.PENDING, loan.amount);
    }

}

File: sources/2015/11/quick-check/qc-scala/src/main/scala/qc/Supervisor.scala
package qc

import qc.State._

object Supervisor {

  // tag::process[]
  def process(loan: Loan) : Loan = loan.amount match { // <1>
    case x if 0   until 201  contains x  => Loan(State.ACCEPTED, loan.amount) // <2>
    case x if 201 until 1001 contains x  => Loan(State.PENDING,  loan.amount) // <3>
    case _                               => Loan(State.REJECTED, loan.amount) // <4>
  }
  // end::process[]

}
